op-service/client/http.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	DefaultTimeoutSeconds = 30
+)
+
+type HTTP interface {
+	Get(ctx context.Context, path string, headers http.Header) (*http.Response, error)
+}
+
+type BasicHTTPClient struct {
+	endpoint string
+	log      log.Logger
+	client   *http.Client
+}
+
+func NewBasicHTTPClient(endpoint string, log log.Logger) *BasicHTTPClient {
+	// Make sure the endpoint ends in trailing slash
+	trimmedEndpoint := strings.TrimSuffix(endpoint, "/") + "/"
+	return &BasicHTTPClient{
+		endpoint: trimmedEndpoint,
+		log:      log,
+		client:   &http.Client{Timeout: DefaultTimeoutSeconds * time.Second},
+	}
+}
+
```

Comment:
It feels more canonical to me to pass the headers in the constructor instead of doing on a per request basis. Are there cases in which its better to do it on a per request basis? Since the endpoint is fixed already. I see that the standard library does it on a per request basis so I am not super opinionated on this. https://pkg.go.dev/net/http#Client.Do

Reply:
For every http verb we will need to implement the same headers handling. You could argue its a premature optimization but the 1 diff between verbs is  the method (`http.MethodGet` in this case)

Reply:
Honestly not a lot of thought went into this api, it's mostly for if/when we want to add more instrumentation around these http calls. But since it's intended to be generic (not just used for querying the L1 Beacon client) I'm worried that requiring headers to be in the constructor might make it too limited? I always have thought of headers as a part of the request vs. the connection (the latter which I'd argue makes sense to mostly configure in the constructor).

Reply:
Ok I am on board with your suggestion, it matches the go std lib api